File: app/src/main/kotlin/com/spacex/ui/rocketDetails/RocketDetailsUIM.java
package com.spacex.ui.rocketDetails;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import com.jjoe64.graphview.series.DataPoint;
import com.jjoe64.graphview.series.LineGraphSeries;
import com.spacex.ui.rocketDetails.launches.RocketLaunchItemUIM;
import java.util.List;

public class RocketDetailsUIM {

  private final int imageResId;
  private final String name;
  private final String description;
  private final List<RocketLaunchItemUIM> launches;
  private final LineGraphSeries<DataPoint> series;

  public RocketDetailsUIM(
      @DrawableRes final int imageResId,
      @NonNull final String name,
      @NonNull final String description,
      @NonNull final List<RocketLaunchItemUIM> launches,
      @NonNull final LineGraphSeries<DataPoint> series) {
    this.name = name;
    this.series = series;
    this.launches = launches;
    this.imageResId = imageResId;
    this.description = description;
  }

  @DrawableRes
  int getImageResId() {
    return imageResId;
  }

  @NonNull
  public String getName() {
    return name;
  }

  @NonNull
  public String getDescription() {
    return description;
  }

  @NonNull
  public List<RocketLaunchItemUIM> getLaunches() {
    return launches;
  }

  @NonNull
  public LineGraphSeries<DataPoint> getChartSeries() {
    return series;
  }
}

File: app/src/main/kotlin/com/spacex/ui/rocketDetails/RocketDetailsUIM.kt
package com.spacex.ui.rocketDetails

import androidx.annotation.DrawableRes
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries
import com.spacex.ui.rocketDetails.launches.RocketLaunchItemUIM

internal data class RocketDetailsUIM(
        @param:DrawableRes @get:DrawableRes
        internal val imageResId: Int,
        val name: String,
        val description: String,
        val launches: List<RocketLaunchItemUIM>,
        val chartSeries: LineGraphSeries<DataPoint?>)
